Erro Oracle - migração versão 3.6.2 para versão 3.7.0

**Descreva o erro**

O erro ocorre quando sobe o Tramita versão 3.6.2 e tenta migrar para a versão 3.7.0, utilizando a versão do SEI 4.0.12.15,utilizando o banco de dados Oracle.

-----

**Para reproduzir**
Passos para reproduzir o comportamento:
1. Tramita versão 3.6.2
2. Migrar para a versão 3.7.0
3.  Versão SEI : 4.0.12.15
4. Banco de Dados - Oracle
5. Veja o erro

-----

**Screenshots**
![Image](https://github.com/user-attachments/assets/0082c511-8ba0-4581-aae2-907a303757f5)


v1: PENAgendamentoRN::processarTarefasPEN

00084 - [InfraOracle->consultarSql] 1 registro(s)
00085 - [InfraOracle->consultarSql] 0,001 s
00086 - [InfraOracle->consultarSql] SELECT infra_agendamento_tarefa.sin_ativo AS sinativo,infra_agendamento_tarefa.comando,infra_agendamento_tarefa.id_infra_agendamento_tarefa AS idinfraagendamentotarefa,infra_agendamento_tarefa.descricao,infra_agendamento_tarefa.sta_periodicidade_execucao AS staperiodicidadeexecucao,infra_agendamento_tarefa.periodicidade_complemento AS periodicidadecomplemento,infra_agendamento_tarefa.parametro,infra_agendamento_tarefa.dth_ultima_execucao AS ultimaexecucao,infra_agendamento_tarefa.dth_ultima_conclusao AS ultimaconclusao,infra_agendamento_tarefa.sin_sucesso AS sinsucesso,infra_agendamento_tarefa.email_erro AS emailerro FROM infra_agendamento_tarefa WHERE infra_agendamento_tarefa.comando=:v1

Bind:
v1: PENAgendamentoRN::processarTarefasPEN

00087 - [InfraOracle->consultarSql] 1 registro(s)
00088 - [InfraOracle->consultarSql] 0,001 s
00089 - [InfraOracle->executarSql] DELETE FROM infra_agendamento_tarefa WHERE infra_agendamento_tarefa.id_infra_agendamento_tarefa=:v1

Bind:
v1: 12

00090 - [InfraOracle->executarSql] 1 registro(s) afetado(s)
00091 - [InfraOracle->executarSql] 0,001 s
00092 - [InfraOracle->consultarSql] SELECT nullable  FROM all_tab_columns WHERE lower(table_name)='md_pen_expedir_lote' AND lower(owner)='sei' and lower(column_name)='id_repositorio_destino'
00093 - [InfraOracle->consultarSql] 1 registro(s)
00094 - [InfraOracle->consultarSql] 0,011 s
00095 - [InfraOracle->executarSql] alter table md_pen_expedir_lote modify id_repositorio_destino number(*,0) null
00096 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00097 - [InfraOracle->executarSql] 0,004 s
00098 - [InfraOracle->consultarSql] SELECT nullable  FROM all_tab_columns WHERE lower(table_name)='md_pen_expedir_lote' AND lower(owner)='sei' and lower(column_name)='str_repositorio_destino'
00099 - [InfraOracle->consultarSql] 1 registro(s)
00100 - [InfraOracle->consultarSql] 0,01 s
00101 - [InfraOracle->executarSql] alter table md_pen_expedir_lote modify str_repositorio_destino varchar2(250 byte) null
00102 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00103 - [InfraOracle->executarSql] 0,003 s
00104 - [InfraOracle->consultarSql] SELECT nullable  FROM all_tab_columns WHERE lower(table_name)='md_pen_expedir_lote' AND lower(owner)='sei' and lower(column_name)='id_repositorio_origem'
00105 - [InfraOracle->consultarSql] 1 registro(s)
00106 - [InfraOracle->consultarSql] 0,01 s
00107 - [InfraOracle->executarSql] alter table md_pen_expedir_lote modify id_repositorio_origem number(*,0) null
00108 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00109 - [InfraOracle->executarSql] 0,003 s
00110 - [InfraOracle->consultarSql] SELECT nullable  FROM all_tab_columns WHERE lower(table_name)='md_pen_expedir_lote' AND lower(owner)='sei' and lower(column_name)='id_unidade_origem'
00111 - [InfraOracle->consultarSql] 1 registro(s)
00112 - [InfraOracle->consultarSql] 0,01 s
00113 - [InfraOracle->executarSql] alter table md_pen_expedir_lote modify id_unidade_origem number(*,0) null
00114 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00115 - [InfraOracle->executarSql] 0,003 s
00116 - [InfraOracle->consultarSql] SELECT nullable  FROM all_tab_columns WHERE lower(table_name)='md_pen_expedir_lote' AND lower(owner)='sei' and lower(column_name)='id_unidade_destino'
00117 - [InfraOracle->consultarSql] 1 registro(s)
00118 - [InfraOracle->consultarSql] 0,01 s
00119 - [InfraOracle->executarSql] alter table md_pen_expedir_lote modify id_unidade_destino number(*,0) null
00120 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00121 - [InfraOracle->executarSql] 0,003 s
00122 - [InfraOracle->consultarSql] SELECT nullable  FROM all_tab_columns WHERE lower(table_name)='md_pen_expedir_lote' AND lower(owner)='sei' and lower(column_name)='str_unidade_destino'
00123 - [InfraOracle->consultarSql] 1 registro(s)
00124 - [InfraOracle->consultarSql] 0,011 s
00125 - [InfraOracle->executarSql] alter table md_pen_expedir_lote modify str_unidade_destino varchar2(250 byte) null
00126 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00127 - [InfraOracle->executarSql] 0,003 s
00128 - [InfraOracle->consultarSql] SELECT nullable  FROM all_tab_columns WHERE lower(table_name)='md_pen_expedir_lote' AND lower(owner)='sei' and lower(column_name)='id_usuario'
00129 - [InfraOracle->consultarSql] 1 registro(s)
00130 - [InfraOracle->consultarSql] 0,01 s
00131 - [InfraOracle->executarSql] alter table md_pen_expedir_lote modify id_usuario number(*,0) null
00132 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00133 - [InfraOracle->executarSql] 0,003 s
00134 - [InfraOracle->consultarSql] SELECT nullable  FROM all_tab_columns WHERE lower(table_name)='md_pen_expedir_lote' AND lower(owner)='sei' and lower(column_name)='id_unidade'
00135 - [InfraOracle->consultarSql] 1 registro(s)
00136 - [InfraOracle->consultarSql] 0,01 s
00137 - [InfraOracle->executarSql] alter table md_pen_expedir_lote modify id_unidade number(*,0) null
00138 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00139 - [InfraOracle->executarSql] 0,003 s
00140 - [InfraOracle->executarSql] alter table md_pen_rel_expedir_lote drop constraint fk_md_pen_rel_expedir_lote
00141 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00142 - [InfraOracle->executarSql] 0,002 s
00143 - [InfraOracle->consultarSql] SELECT lower(column_name) as column_name, nullable as is_nullable, lower(data_type) as data_type, char_length as character_maximum_length, data_precision as numeric_precision, data_scale as numeric_scale
          FROM all_tab_columns
          WHERE lower(table_name)='md_pen_expedir_lote' AND lower(owner)='sei'
00144 - [InfraOracle->consultarSql] 10 registro(s)
00145 - [InfraOracle->consultarSql] 0,046 s
00146 - [InfraOracle->executarSql] ALTER TABLE md_pen_expedir_lote RENAME COLUMN id_lote TO id_bloco_processo
00147 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00148 - [InfraOracle->executarSql] 0,005 s
00149 - [InfraOracle->executarSql] alter table md_pen_expedir_lote add dth_atualizado date NULL
00150 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00151 - [InfraOracle->executarSql] 0,004 s
00152 - [InfraOracle->executarSql] alter table md_pen_expedir_lote add dth_envio date NULL
00153 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00154 - [InfraOracle->executarSql] 0,003 s
00155 - [InfraOracle->executarSql] alter table md_pen_expedir_lote add id_protocolo number(*,0) NULL
00156 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00157 - [InfraOracle->executarSql] 0,003 s
00158 - [InfraOracle->executarSql] alter table md_pen_expedir_lote add id_bloco number(*,0) NULL
00159 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00160 - [InfraOracle->executarSql] 0,004 s
00161 - [InfraOracle->executarSql] alter table md_pen_expedir_lote add sequencia number(*,0) NULL
00162 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00163 - [InfraOracle->executarSql] 0,003 s
00164 - [InfraOracle->executarSql] alter table md_pen_expedir_lote add id_andamento number(*,0) NULL
00165 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00166 - [InfraOracle->executarSql] 0,004 s
00167 - [InfraOracle->executarSql] alter table md_pen_expedir_lote add id_atividade_expedicao varchar2(4000 byte) NULL
00168 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00169 - [InfraOracle->executarSql] 0,003 s
00170 - [InfraOracle->executarSql] alter table md_pen_expedir_lote add tentativas number(*,0) NULL
00171 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00172 - [InfraOracle->executarSql] 0,003 s
00173 - [InfraOracle->executarSql] alter table md_pen_expedir_lote drop constraint fk_bloco_protocolo
00174 - [Infra->infraGerarExcecao] 10: Warning: /opt/infra/infra_php/InfraOracle.php linha:632.
oci_execute(): ORA-02443: Cannot drop constraint  - nonexistent constraint

00175 - [InfraOracle->executarSql] alter table md_pen_rel_expedir_lote drop constraint fk_md_pen_rel_expedir_lote
00176 - [Infra->infraGerarExcecao] 10: Warning: /opt/infra/infra_php/InfraOracle.php linha:632.
oci_execute(): ORA-02443: Cannot drop constraint  - nonexistent constraint

00177 - [InfraOracle->executarSql] alter table md_pen_bloco_protocolo drop constraint fk_bloco_protocolo
00178 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00179 - [InfraOracle->executarSql] 0,002 s
00180 - [InfraOracle->consultarSql] select lower(table_name) as table_name from all_tables where lower(owner)='sei' and lower(table_name)='md_pen_expedir_lote' order by table_name asc
00181 - [InfraOracle->consultarSql] 1 registro(s)
00182 - [InfraOracle->consultarSql] 0,071 s
00183 - [InfraOracle->executarSql] RENAME md_pen_expedir_lote TO md_pen_bloco_processo
00184 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00185 - [InfraOracle->executarSql] 0,015 s
00186 - [InfraOracle->consultarSql] select lower(constraint_name) as constraint_name, decode(constraint_type, 'P', 'primary key', 'R','foreign key') as constraint_type, lower(table_name) as table_name from all_constraints where lower(constraint_type) in ('p','r')
              AND lower(owner)='sei' and lower(table_name)='md_pen_bloco_processo' order by table_name, constraint_name
00187 - [InfraOracle->consultarSql] 3 registro(s)
00188 - [InfraOracle->consultarSql] 0,191 s
00189 - [InfraOracle->executarSql] alter table md_pen_bloco_processo add constraint fk_md_pen_bloco_proc_procedi foreign key (id_protocolo) references protocolo (id_protocolo)
00190 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00191 - [InfraOracle->executarSql] 0,009 s
00192 - [InfraOracle->consultarSql] select lower(constraint_name) as constraint_name, decode(constraint_type, 'P', 'primary key', 'R','foreign key') as constraint_type, lower(table_name) as table_name from all_constraints where lower(constraint_type) in ('p','r')
              AND lower(owner)='sei' and lower(table_name)='md_pen_bloco_processo' order by table_name, constraint_name
00193 - [InfraOracle->consultarSql] 4 registro(s)
00194 - [InfraOracle->consultarSql] 0,204 s
00195 - [InfraOracle->executarSql] alter table md_pen_bloco_processo add constraint fk_md_pen_bloco_processo_bl foreign key (id_bloco) references md_pen_bloco (id)
00196 - [InfraOracle->executarSql] 0 registro(s) afetado(s)
00197 - [InfraOracle->executarSql] 0,009 s
00198 - [InfraOracle->executarSql] alter table md_pen_bloco add ordem number(*,0) NOT NULL
00199 - [Infra->infraGerarExcecao] 10: Warning: /opt/infra/infra_php/InfraOracle.php linha:632.
oci_execute(): ORA-01758: table must be empty to add mandatory (NOT NULL) column

Descri��o:
Erro atualizando VERSAO: Exception: 'Warning: /opt/infra/infra_php/InfraOracle.php linha:632.
oci_execute(): ORA-01758: table must be empty to add mandatory (NOT NULL) column
' in /opt/infra/infra_php/Infra.php:248
Stack trace:
#0 [internal function]: infraGerarExcecao(2, 'oci_execute(): ...', '/opt/infra/infr...', 632, Array)
#1 /opt/infra/infra_php/InfraOracle.php(632): oci_execute(Resource id #242, 32)
#2 /opt/infra/infra_php/InfraMetaBD.php(776): InfraOracle->executarSql('alter table md_...')
#3 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(2904): InfraMetaBD->adicionarColuna('md_pen_bloco', 'ordem', 'number(*,0)', 'NOT NULL')
#4 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(285): PenAtualizarSeiRN->instalarV3070()
#5 /opt/infra/infra_php/InfraRN.php(149): PenAtualizarSeiRN->atualizarVersaoConectado()
#6 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(45): InfraRN->__call('atualizarVersao', Array)
#7 /opt/infra/infra_php/InfraScriptVersao.php(394): VersaoSei4RN->atualizarVersaoCompatibilidade('3.6.2')
#8 /opt/infra/infra_php/InfraRN.php(151): InfraScriptVersao->atualizarVersaoInternoConectado(Object(BancoSEI))
#9 /opt/infra/infra_php/InfraScriptVersao.php(233): InfraRN->__call('atualizarVersao...', Array)
#10 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(3106): InfraScriptVersao->atualizarVersao()
#11 {main}

Mensagem:
Erro atualizando VERSAO: Exception: 'Warning: /opt/infra/infra_php/InfraOracle.php linha:632.
oci_execute(): ORA-01758: table must be empty to add mandatory (NOT NULL) column
' in /opt/infra/infra_php/Infra.php:248
Stack trace:
#0 [internal function]: infraGerarExcecao(2, 'oci_execute(): ...', '/opt/infra/infr...', 632, Array)
#1 /opt/infra/infra_php/InfraOracle.php(632): oci_execute(Resource id #242, 32)
#2 /opt/infra/infra_php/InfraMetaBD.php(776): InfraOracle->executarSql('alter table md_...')
#3 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(2904): InfraMetaBD->adicionarColuna('md_pen_bloco', 'ordem', 'number(*,0)', 'NOT NULL')
#4 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(285): PenAtualizarSeiRN->instalarV3070()
#5 /opt/infra/infra_php/InfraRN.php(149): PenAtualizarSeiRN->atualizarVersaoConectado()
#6 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(45): InfraRN->__call('atualizarVersao', Array)
#7 /opt/infra/infra_php/InfraScriptVersao.php(394): VersaoSei4RN->atualizarVersaoCompatibilidade('3.6.2')
#8 /opt/infra/infra_php/InfraRN.php(151): InfraScriptVersao->atualizarVersaoInternoConectado(Object(BancoSEI))
#9 /opt/infra/infra_php/InfraScriptVersao.php(233): InfraRN->__call('atualizarVersao...', Array)
#10 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(3106): InfraScriptVersao->atualizarVersao()
#11 {main}

Trilha de Processamento:
#0 /opt/infra/infra_php/InfraScriptVersao.php(235): InfraScript->processarErro('ERRO ATUALIZAND...', Object(InfraException))
#1 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(3106): InfraScriptVersao->atualizarVersao()
#2 {main}

Descri��o Original:
Exception: 'Warning: /opt/infra/infra_php/InfraOracle.php linha:632.
oci_execute(): ORA-01758: table must be empty to add mandatory (NOT NULL) column
' in /opt/infra/infra_php/Infra.php:248
Stack trace:
#0 [internal function]: infraGerarExcecao(2, 'oci_execute(): ...', '/opt/infra/infr...', 632, Array)
#1 /opt/infra/infra_php/InfraOracle.php(632): oci_execute(Resource id #242, 32)
#2 /opt/infra/infra_php/InfraMetaBD.php(776): InfraOracle->executarSql('alter table md_...')
#3 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(2904): InfraMetaBD->adicionarColuna('md_pen_bloco', 'ordem', 'number(*,0)', 'NOT NULL')
#4 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(285): PenAtualizarSeiRN->instalarV3070()
#5 /opt/infra/infra_php/InfraRN.php(149): PenAtualizarSeiRN->atualizarVersaoConectado()
#6 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(45): InfraRN->__call('atualizarVersao', Array)
#7 /opt/infra/infra_php/InfraScriptVersao.php(394): VersaoSei4RN->atualizarVersaoCompatibilidade('3.6.2')
#8 /opt/infra/infra_php/InfraRN.php(151): InfraScriptVersao->atualizarVersaoInternoConectado(Object(BancoSEI))
#9 /opt/infra/infra_php/InfraScriptVersao.php(233): InfraRN->__call('atualizarVersao...', Array)
#10 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(3106): InfraScriptVersao->atualizarVersao()
#11 {main}

Trilha de Processamento [Original:]([url](url))
\n0 [internal function]: infraGerarExcecao(2, 'oci_execute(): ...', '/opt/infra/infr...', 632, Array)
\n1 /opt/infra/infra_php/InfraOracle.php(632): oci_execute(Resource id \n242, 32)
\n2 /opt/infra/infra_php/InfraMetaBD.php(776): InfraOracle->executarSql('alter table md_...')
\n3 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(2904): InfraMetaBD->adicionarColuna('md_pen_bloco', 'ordem', 'number(*,0)', 'NOT NULL')
\n4 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(285): PenAtualizarSeiRN->instalarV3070()
\n5 /opt/infra/infra_php/InfraRN.php(149): PenAtualizarSeiRN->atualizarVersaoConectado()
\n6 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(45): InfraRN->__call('atualizarVersao', Array)
\n7 /opt/infra/infra_php/InfraScriptVersao.php(394): VersaoSei4RN->atualizarVersaoCompatibilidade('3.6.2')
\n8 /opt/infra/infra_php/InfraRN.php(151): InfraScriptVersao->atualizarVersaoInternoConectado(Object(BancoSEI))
\n9 /opt/infra/infra_php/InfraScriptVersao.php(233): InfraRN->__call('atualizarVersao...', Array)
\n10 /opt/sei/scripts/mod-pen/sei_atualizar_versao_modulo_pen.php(3106): InfraScriptVersao->atualizarVersao()
\n11 {main}

make: *** [Makefile:147: install] Error 1

--- a/src/scripts/sei_atualizar_versao_modulo_pen.php
+++ b/src/scripts/sei_atualizar_versao_modulo_pen.php
@@ -2801,6 +2801,7 @@ class PenAtualizarSeiRN extends PenAtualizadorRN
   {
     // Criação da tabela restrição
     $objMetaRestricaoBD = $this->objMeta;
+    $SNULLO = BancoSEI::getInstance() instanceof InfraPostgreSql ? 'null' : PenMetaBD::SNULLO;
 
     // Remoção de coluna sin_padrao da tabela md_pen_rel_doc_map_enviado
     $this->logar("CRIANDO TABELA DE CONFIGURACAO PARA RESTRICAO ");
@@ -2810,10 +2811,10 @@ class PenAtualizarSeiRN extends PenAtualizadorRN
         'id' => array($objMetaRestricaoBD->tipoNumeroGrande(), PenMetaBD::NNULLO),
         'id_unidade' => array($objMetaRestricaoBD->tipoNumero(), PenMetaBD::NNULLO),
         'id_unidade_rh' => array($objMetaRestricaoBD->tipoNumeroGrande(), PenMetaBD::NNULLO),
-        'id_unidade_restricao' => array($objMetaRestricaoBD->tipoNumeroGrande(), PenMetaBD::SNULLO),
-        'nome_unidade_restricao' => array($objMetaRestricaoBD->tipoTextoVariavel(255), PenMetaBD::SNULLO),
-        'id_unidade_rh_restricao' => array($objMetaRestricaoBD->tipoNumeroGrande(), PenMetaBD::SNULLO),
-        'nome_unidade_rh_restricao' => array($objMetaRestricaoBD->tipoTextoVariavel(255), PenMetaBD::SNULLO),
+        'id_unidade_restricao' => array($objMetaRestricaoBD->tipoNumeroGrande(), PenMetaBD::NNULLO),
+        'nome_unidade_restricao' => array($objMetaRestricaoBD->tipoTextoVariavel(255), $SNULLO),
+        'id_unidade_rh_restricao' => array($objMetaRestricaoBD->tipoNumeroGrande(), PenMetaBD::NNULLO),
+        'nome_unidade_rh_restricao' => array($objMetaRestricaoBD->tipoTextoVariavel(255), $SNULLO),
       ),
       'pk' => array('cols' => array('id')),
       'fks' => array(
@@ -2854,7 +2855,6 @@ class PenAtualizarSeiRN extends PenAtualizadorRN
     }
 
     // Alterar colunas em md_pen_expedir_lote
-    $SNULLO = BancoSEI::getInstance() instanceof InfraPostgreSql ? 'null' : PenMetaBD::SNULLO;
     $objMetaBD->alterarColuna('md_pen_expedir_lote', 'id_repositorio_destino', $objMetaBD->tipoNumero(), $SNULLO);
     $objMetaBD->alterarColuna('md_pen_expedir_lote', 'str_repositorio_destino', $objMetaBD->tipoTextoVariavel(250), $SNULLO);
     $objMetaBD->alterarColuna('md_pen_expedir_lote', 'id_repositorio_origem', $objMetaBD->tipoNumero(), $SNULLO);
@@ -2875,16 +2875,16 @@ class PenAtualizarSeiRN extends PenAtualizadorRN
     }
 
     // Adicionar coluna de atualização do registro
-    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'dth_atualizado', $objMetaBD->tipoDataHora(), PenMetaBD::SNULLO);
-    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'dth_envio', $objMetaBD->tipoDataHora(), PenMetaBD::SNULLO);
+    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'dth_atualizado', $objMetaBD->tipoDataHora(), $SNULLO);
+    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'dth_envio', $objMetaBD->tipoDataHora(), $SNULLO);
 
     // Adicionar campos extrar para a tabela md_pen_expedir_lote
-    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'id_protocolo', $objMetaBD->tipoNumeroGrande(10), PenMetaBD::SNULLO);
-    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'id_bloco', $objMetaBD->tipoNumero(10), PenMetaBD::SNULLO);
-    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'sequencia', $objMetaBD->tipoNumero(10), PenMetaBD::SNULLO);
-    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'id_andamento', $objMetaBD->tipoNumero(11), PenMetaBD::SNULLO);
-    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'id_atividade_expedicao', $objMetaBD->tipoTextoVariavel(4000), PenMetaBD::SNULLO);
-    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'tentativas', $objMetaBD->tipoNumero(), PenMetaBD::SNULLO);
+    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'id_protocolo', $objMetaBD->tipoNumeroGrande(10), PenMetaBD::NNULLO);
+    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'id_bloco', $objMetaBD->tipoNumero(10), $SNULLO);
+    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'sequencia', $objMetaBD->tipoNumero(10), $SNULLO);
+    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'id_andamento', $objMetaBD->tipoNumero(11), $SNULLO);
+    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'id_atividade_expedicao', $objMetaBD->tipoTextoVariavel(4000), $SNULLO);
+    $objMetaBD->adicionarColuna('md_pen_expedir_lote', 'tentativas', $objMetaBD->tipoNumero(), $SNULLO);
 
     $this->excluirChaveEstrangeira("md_pen_expedir_lote", "fk_bloco_protocolo", true);
     $this->excluirChaveEstrangeira("md_pen_rel_expedir_lote", "fk_md_pen_rel_expedir_lote", true);
@@ -2899,9 +2899,6 @@ class PenAtualizarSeiRN extends PenAtualizadorRN
 
     $objMetaBD->adicionarChaveEstrangeira("fk_md_pen_bloco_proc_procedi", "md_pen_bloco_processo", array('id_protocolo'), "protocolo", array('id_protocolo'), false);
     $objMetaBD->adicionarChaveEstrangeira("fk_md_pen_bloco_processo_bl", "md_pen_bloco_processo", array('id_bloco'), "md_pen_bloco", array('id'), false);
-
-    //Adicionar coluna para ordenar blocos por unidade 
-    $objMetaBD->adicionarColuna('md_pen_bloco', 'ordem', $objMetaBD->tipoNumero(10), PenMetaBD::NNULLO);
 
     $objInfraSequenciaRN = new InfraSequenciaRN();
     $objInfraSequenciaDTO = new InfraSequenciaDTO();
@@ -2918,12 +2915,13 @@ class PenAtualizarSeiRN extends PenAtualizadorRN
     // Atualizar md_pen_bloco_processo->ordem para 1
     $objInfraBanco->executarSql('delete from md_pen_bloco_processo');   
 
-    $objInfraBanco->executarSql('update md_pen_bloco set ordem=1');
-
     $objMetaBD->alterarColuna('md_pen_bloco_processo', 'id_bloco', $objMetaBD->tipoNumero(10), PenMetaBD::NNULLO);
 
     // Excluir bloco legados
     $this->limparBlocos();
+
+    //Adicionar coluna para ordenar blocos por unidade 
+    $objMetaBD->adicionarColuna('md_pen_bloco', 'ordem', $objMetaBD->tipoNumero(10), PenMetaBD::NNULLO);
 
     $tabelas = ['md_pen_bloco_protocolo',
                 'md_pen_seq_bloco_protocolo',
@@ -3137,4 +3135,3 @@ try {
 
   exit(1);
 }
-

